resources/puppetlabs/lein-ezbake/template/global/ext/cli/start.erb

```diff
@@ -26,19 +64,33 @@ fi
 
 ${JAVA_BIN} ${JAVA_ARGS} -Djava.security.egd=/dev/urandom -cp ${INSTALL_DIR}/<%= EZBake::Config[:uberjar_name] %> clojure.main -m <%= EZBake::Config[:main_namespace] %> --config "${CONFIG}" -b "${BOOTSTRAP_CONFIG}" &
 
+pid=$!
+trap terminate_java_process SIGHUP SIGINT SIGTERM
+write_pid_file $pid
+
 cur=$(head -n 1 "$restartfile")
 initial=$cur
 
+timeout=$start_timeout
+set +e
 while [ "$cur" == "$initial" ] ;do
+    kill -0 $pid >/dev/null 2>&1
+    if [ $? != 0 ]; then
+      rm -f "$PIDFILE"
+      echo "Background process $pid exited before start had completed" 1>&2
+      exit 1
+    fi
+
     sleep 1
     cur=$(head -n 1 "$restartfile")
 
     ((timeout--))
     if [ "$timeout" = 0 ]; then
-        echo "Startup timed out after <%= EZBake::Config[:start_timeout] %> seconds" 1>&2
+        echo "Startup timed out after $start_timeout seconds" 1>&2
         exit 1
     fi
 done
+set -e
 
-find_my_pid
-exit 0
\ No newline at end of file
+write_pid_file $pid
```

Comment:
This probably seems redundant with the `write_pid_file` line above - close to where the background process is launched.  In doing some edge-case testing around concurrent restart attempts, I found that for some service frameworks that the pidfile could be removed if a separate script instance was terminated, leading it to not be present after this point even though the service had successfully started.  This just seemed like a good sanity check.  Should be harmless, essentially a no-op in most cases.